src/Microsoft.AspNetCore.Http.Abstractions/Routing/IEndpointFeature.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Features
+{
+    /// <summary>
+    /// A feature interface for endpoint routing. Use <see cref="HttpContext.Features"/>
+    /// to access an instance associated with the current request.
+    /// </summary>
+    public interface IEndpointFeature
+    {
+        /// <summary>
+        /// Gets or sets the selected <see cref="Routing.Endpoint"/> for the current
+        /// request.
+        /// </summary>
+        Endpoint Endpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets a delegate that can be used to invoke the current
+        /// <see cref="Routing.Endpoint"/>.
+        /// </summary>
+        Func<RequestDelegate, RequestDelegate> Invoker { get; set; }
```

Comment:
I think we should try to decide on this before merging to the best of our knowledge. I'm not sure that we understand any reason why this takes 'next' and we're paying a perf and complexity cost for it. IIRC we did what it be do largely as a hedge.

Reply:
https://github.com/aspnet/Routing/pull/707